data-prepper-plugins/saas-source-plugins/saas-crawler/src/main/java/org/opensearch/dataprepper/plugins/source/saas_crawler/base/SaasSourcePlugin.java

```diff
@@ -0,0 +1,120 @@
+package org.opensearch.dataprepper.plugins.source.saas_crawler.base;
+
+
+import lombok.Getter;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSetManager;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.codec.ByteDecoder;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.plugin.PluginFactory;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+import org.opensearch.dataprepper.model.source.coordinator.SourcePartitionStoreItem;
+import org.opensearch.dataprepper.model.source.coordinator.enhanced.EnhancedSourceCoordinator;
+import org.opensearch.dataprepper.model.source.coordinator.enhanced.EnhancedSourcePartition;
+import org.opensearch.dataprepper.model.source.coordinator.enhanced.UsesEnhancedSourceCoordination;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.coordination.PartitionFactory;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.coordination.partition.LeaderPartition;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.coordination.scheduler.LeaderScheduler;
+import org.opensearch.dataprepper.plugins.source.saas_crawler.coordination.scheduler.WorkerScheduler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Function;
+
+
+
+/**
+ * JiraConnector connector entry point.
+ */
+
+public abstract class SaasSourcePlugin implements Source<Record<Event>>, UsesEnhancedSourceCoordination {
+
+
+  private static final Logger log = LoggerFactory.getLogger(SaasSourcePlugin.class);
+  private final PluginMetrics pluginMetrics;
+  private final PluginFactory pluginFactory;
+
+  private final AcknowledgementSetManager acknowledgementSetManager;
+
+  private final ExecutorService executorService;
+
+  private EnhancedSourceCoordinator coordinator;
+
+  @Getter
+  private final SaasSourceConfig sourceConfig;
+
+  private Buffer<Record<Event>> buffer;
+  private final Crawler crawler;
+
+
+  @DataPrepperPluginConstructor
```

Comment:
I do not think you use `@DataPrepperPluginConstructor` for abstract source classes. see `./data-prepper-plugins/http-source-common/src/main/java/org/opensearch/dataprepper/http/BaseHttpSource.java`

Reply:
Agree. No use of this annotation here. Removed it.